File: CS/1773_Count_Items_Matching_a_Rule.cs
public class Solution {
    public int CountMatches(IList<IList<string>> items, string ruleKey, string ruleValue) {
        Dictionary<string, int> dic = new Dictionary<string, int>();
        dic.Add("type", 0);
        dic.Add("color", 1);
        dic.Add("name", 2);
        int result = 0;
        for(int i = 0; i < items.Count; i++){
            if(items[i][dic[ruleKey]] == ruleValue){
                result++;
            }
        }
        return result;
    }
}

File: Cpp/1773_Count_Items_Matching_a_Rule.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        int result = 0;
        map<string, int> m;
        m["type"] = 0;
        m["color"] = 1;
        m["name"] = 2;
        int i = m[ruleKey];
        /*
        for(vector<string> item : items){
            if(item[i] == ruleValue){
                result++;
            }
        }
        */
        return count_if(items.begin(), 
                        items.end(), 
                        [i, ruleValue](auto item)
                        { return item[i] == ruleValue; });
        //return result;
    }
};
